internal/ls/utilities.go

```diff
@@ -724,3 +1525,106 @@ func getAllSuperTypeNodes(node *ast.Node) []*ast.TypeNode {
 	}
 	return nil
 }
+
+func getParentSymbolsOfPropertyAccess(location *ast.Node, symbol *ast.Symbol, ch *checker.Checker) []*ast.Symbol {
+	propertyAccessExpression := core.IfElse(isRightSideOfPropertyAccess(location), location.Parent, nil)
+	if propertyAccessExpression == nil {
+		return nil
+	}
+
+	lhsType := ch.GetTypeAtLocation(propertyAccessExpression.Expression())
+	if lhsType == nil {
+		return nil
+	}
+
+	res := core.MapNonNil(
+		core.IfElse(lhsType.Flags() != 0, lhsType.Types(), core.IfElse(lhsType.Symbol() == symbol.Parent, nil, []*checker.Type{lhsType})),
```

Comment:
this feels like it does too much work for a pair of IfElses, though it would be much longer to write out as if/elses.